semgrep/semgrep/evaluation.py

```diff
@@ -56,6 +55,68 @@ def get_re_range_matches(
     return result
 
 
+def compare_range_match(
+    metavariable: str,
+    comparison: str,
+    strip: Optional[bool],
+    base: Optional[int],
+    content: str,
+) -> bool:
+
+    if strip:
+        content = content.strip("\"'`")
+
+    try:
+        # Assume float data if "." in content
+        if "." in content:
+            converted = float(content)
+        else:
+            if base:
+                converted = int(content, base=base)
+            else:
+                converted = int(content)
+    except ValueError:
+        logger.debug(
+            f"metavariable '{metavariable}' incorrect comparison type '{content}'"
+        )
+        return False
+
+    return metavariable_comparison(metavariable, comparison, converted)
+
+
+def get_comparison_range_matches(
+    metavariable: str,
+    comparison: str,
+    strip: Optional[bool],
+    base: Optional[int],
+    ranges: Set[Range],
+    pattern_matches: List[PatternMatch],
+) -> Set[Range]:
+
+    result: Set[Range] = set()
+    for _range in ranges:
+        if metavariable not in _range.vars:
+            logger.debug(f"metavariable '{metavariable}' missing in range '{_range}'")
+            continue
+
+        any_matching_ranges = any(
+            pm.range == _range
+            and metavariable in pm.metavars
+            and compare_range_match(
+                metavariable,
+                comparison,
+                strip,
+                base,
+                pm.metavars[metavariable]["abstract_content"],
```

Comment:
@brendongo might get rid of the use of abstract_content at some point, which is quite fragile.

Reply:
It's used in quite a few places, so we'll have to tread carefully if/when we do.